dirs/dirs.go

```diff
@@ -163,8 +163,12 @@ var (
 )
 
 type SnapDirOptions struct {
-	// HiddenSnapDataDir determines if the snaps' data is in ~/.snap/data instead of ~/snap
-	HiddenSnapDataDir bool
```

Comment:
I think this change is confusing. We should have a different structure if needed about the state of the dir vs what is consumed by the snap code.